antarest/study/service.py

```diff
@@ -2518,3 +2543,28 @@ def get_matrix_with_index_and_header(
         )
 
         return df_matrix
+
+    def assert_no_cluster_referenced_in_bcs(self, study: Study, area_id: str, cluster_ids: t.Sequence[str]) -> None:
+        """
+        Check that no cluster is referenced in a binding constraint otherwise raise an ClusterDeletionNotAllowed Exception
+
+        Args:
+            study: input study
+            area_id: area ID to be checked
+            cluster_ids: cluster IDs to be checked
+
+        Returns:
+
+        """
+
+        for cluster_id in cluster_ids:
+            referencing_binding_constraints = self.binding_constraint_manager.get_binding_constraints(
+                study, ConstraintFilters(cluster_id=f"{area_id}.{cluster_id}")
+            )[:MAX_BINDING_CONSTRAINTS_TO_DISPLAY]
+            if referencing_binding_constraints:
```

Comment:
You could perhaps copy this code inside a method that you would use inside the 3 exceptions. 

This way you will avoid duplicating this "ugly" code.